### ISSUE TITLE
Issues running testfile trivial.py

### ISSUE THREAD

**Issue by christianfrech** [spparks/spparks#20]
Hello,

I have successfully built SPPARKS as a shared library to use with Python, and I am testing it with the example files. I am running the trivial.py test file with the ising model input.  The trivial.py test file is listed below:

```
#!/usr/local/bin/python -i
# preceeding line should have path for Python on your machine

# trivial.py
# Purpose: run a SPPARKS input script via Python
# Syntax:  trivial.py in.spparks
#          in.spparks = SPPARKS input script

import sys

# parse command line

argv = sys.argv
if len(argv) != 2:
  print "Syntax: trivial.py in.spparks"
  sys.exit()

infile = sys.argv[1]

me = 0
# uncomment if running in parallel via Pypar
#import pypar
#me = pypar.rank()
#nprocs = pypar.size()

from spparks import spparks
spk = spparks()

# run infile all at once

#spk.file(infile)

# run infile one line at a time

lines = open(infile,'r').readlines()
for line in lines: spk.command(line)  **#### FILE BREAKS HERE ###**

# uncomment if running in parallel via Pypar
#print "Proc %d out of %d procs has" % (me,nprocs), spk
#pypar.finalize()
```

Everything is working until the command "for line in lines: spk.command(line)", where the following error is raised:
```
ERROR: App_style specific command before app_style set (../input.cpp:447)
```
I have looked through the input.cpp file extensively and have no clue how to fix this problem. The issue is listed as self-explanatory on the documentation.

Thanks!

**Comment by sjplimp** [comment]
I think the issue was incompatibility of the Python interface with Python3.  A new PR, which should be merged today, makes the spparks.py interface Python3 ready.  This also affects how strings are passed to the SPPARKS lib via ctypes in spparks.py.
The appended Python3 version of your trivial.py now works on my box.

#!/usr/local/bin/python -i
# preceeding line should have path for Python on your machine

# trivial.py
# Purpose: run a SPPARKS input script via Python
# Syntax:  trivial.py in.spparks
#          in.spparks = SPPARKS input script

import sys

# parse command line

argv = sys.argv
if len(argv) != 2:
  print("Syntax: trivial.py in.spparks")
  sys.exit()

infile = sys.argv[1]

me = 0
# uncomment if running in parallel via Pypar
#import pypar
#me = pypar.rank()
#nprocs = pypar.size()

from spparks import spparks
spk = spparks()

# run infile all at once

#spk.file(infile)

# run infile one line at a time

lines = open(infile,'r').readlines()
for line in lines: spk.command(line)

# uncomment if running in parallel via Pypar
#print "Proc %d out of %d procs has" % (me,nprocs), spk
#pypar.finalize()

**Comment by zhfzzz** [comment]
![image](https://user-images.githubusercontent.com/75823225/234776979-c847b15b-cd64-4551-ba16-68467a121c45.png)
Dear Prof. /Dr. sjplimp and jamitch-snl
I encountered difficulties and read https://spparks.github.io/doc/dump_image.html This website address, but I didn't understand it. I tried various methods but still couldn't solve it. I sincerely request your help. 
Thank you very much.
